### DISCUSSION TITLE
Row grouping

### DISCUSSION THREAD

**Discussion by sarbull** [icflorescu/mantine-datatable#310] [Ideas]
**Is your feature request related to a problem? Please describe.**
Feature request.

**Describe the solution you'd like**
![image](https://user-images.githubusercontent.com/1864112/205742822-88358906-e99c-4583-b5ba-12cec13ace82.png)

**Describe alternatives you've considered**
https://www.ag-grid.com/angular-data-grid/grouping-group-rows/, but this is enterprise

**Additional context**
I can help with the implementation, but i need guidance. I am using mantine for this [0] awesome project.

[0] - https://github.com/odpi/egeria-ui

**Comment by icflorescu** [comment]
I have no idea how this could be best implemented, but I think the [row expansion feature](https://icflorescu.github.io/mantine-datatable/examples/expanding-rows) could be a good starting point. Have a look at [DataTableRowExpansion.tsx](https://github.com/icflorescu/mantine-datatable/blob/main/package/DataTableRowExpansion.tsx) and [this hook](https://github.com/icflorescu/mantine-datatable/blob/main/package/hooks.ts#L115).
Can't use the Mantine `Collapse` component in this case, though, because you'd want your content to be rendered as `<tr />`s in the end....

**Comment by Rigel772** [comment]
Mayby how it is done in DevExtreme could be inspiration...
https://js.devexpress.com/Demos/WidgetsGallery/Demo/DataGrid/RecordGrouping/React/Light/
